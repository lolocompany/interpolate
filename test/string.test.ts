import { it, describe } from 'vitest';

import { interpolate } from '../src/main';

const params = '{x.y}';

describe('Array', () => {
  it('Empty context', ({ expect }) => {
    const ctx = {};
    expect(interpolate(ctx, params)).toMatchSnapshot();
  });

  it('Exact match context', ({ expect }) => {
    const ctx = {
      x: { y: 'Hello' },
    };
    expect(interpolate(ctx, params)).toMatchSnapshot();
  });

  it('No match context', ({ expect }) => {
    const ctx = {
      a: { b: 'Hello' },
    };
    expect(interpolate(ctx, params)).toMatchSnapshot();
  });

  it('Full context', ({ expect }) => {
    const ctx = {
      x: { y: 'Hello' },
      a: { b: 'Hello' },
    };
    expect(interpolate(ctx, params)).toMatchSnapshot();
  });
});
