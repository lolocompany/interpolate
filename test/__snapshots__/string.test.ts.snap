// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Array > Empty context 1`] = `"{x.y}"`;

exports[`Array > Exact match context 1`] = `"Hello"`;

exports[`Array > Full context 1`] = `"Hello"`;

exports[`Array > No match context 1`] = `"{x.y}"`;
